index.bs

```diff
@@ -5770,6 +5784,122 @@ The [=remote end steps=] with |command parameters| are:
 
 </div>
 
+#### The emulation.setOrientationOverride Command ####  {#command-emulation-setOrientationOverride}
+
+The <dfn export for=commands>emulation.setOrientationOverride</dfn> command emulates
+[=screen orientation=] of the given top-level traversables or user contexts.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      emulation.SetOrientationOverride = (
+        method: "emulation.setOrientationOverride",
+        params: emulation.SetOrientationOverrideParameters
+      )
+
+      emulation.OrientationAngle = 0 / 90 / 180 / 270
+
+      emulation.OrientationType = "portrait-primary" / "portrait-secondary" / "landscape-primary" / "landscape-secondary"
+
+      emulation.Orientation = {
+        angle: emulation.OrientationAngle,
+        type: emulation.OrientationType
+      }
+
+      emulation.SetOrientationOverrideParameters = {
+        orientation: emulation.Orientation / null,
+        ? contexts: [+browsingContext.BrowsingContext],
+        ? userContexts: [+browser.UserContext],
+      }
+    </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <code>
+     EmptyResult
+    </code>
+   </dd>
+</dl>
+
+<div algorithm>
+
+To <dfn>set emulated orientation</dfn> given |emulated orientation|:
+
+Issue: Move this algorithm to screen orientation specification.
+
+1. If |emulated orientation| is null:
+
+   1. Set |top-level navigable|'s [=current orientation angle=] to
+      implementation-defined default.
+
+   1. Set |top-level navigable|'s [=current orientation type=] to
+      implementation-defined default.
+
+1. Otherwise:
+
+   1. Set [=current orientation angle=] to
+      |emulated orientation|["<code>orientationAngle</code>"].
+
+   1. Set [=current orientation type=] to
+      |emulated orientation|["<code>orientationType</code>"].
+
+1. Run the [=screen orientation change steps=] with the |top-level navigable|'s
+   [=active document=].
+
+</div>
+
+<div algorithm="remote end steps for emulation.setOrientationOverride">
+
+The [=remote end steps=] with |command parameters| are:
+
+1. If the implementation is unable to adjust the [=screen orientations=]
+   parameters with the given |command parameters| for any reason, return
+   [=error=] with [=error code=] [=unsupported operation=].
+
+1. If |command parameters| [=map/contains=] "<code>userContexts</code>"
```

Comment:
As discussed, I think from here we could define a re-usable algorithm for a command applicable to userContexts with contexts, starting from here, everything seems exactly the same as for other similar commands and the variable bits are:

1) per-user-context steps like `[=map/Set=] [=orientation overrides map=][|user context|] to |emulated orientation|.`
2) per-navigable steps like `[=Set emulated orientation=] with |navigable| and |emulated orientation|`.

It would be great to try to refactor the spec and de-duplicate these common steps.